pkg/build/api/types.go

```diff
@@ -476,3 +479,12 @@ type BuildLogOptions struct {
 	// is not available yet. Otherwise the server will wait until the build has started.
 	NoWait bool
 }
+
+// SecretSpec specifies a secret to be included in a build pod and its corresponding mount point
+type SecretSpec struct {
```

Comment:
While I want to be able to mount secrets into my builders somehow, I don't
think the API design would be the same (mountPath in particular), so let's
limit this to custom strategy.

I do have the general concern that we are replicating a PodTemplate here
and in ExecNewPod by selectively bringing in the attributes we need - at
some point, we should be defining a new sub type which is explicitly
shared.  It does not have to be done in this pull, we can always migrate
away from it in the future.

If this is moved to custom strategy and the name is more appropriate API is
LGTM

On Fri, Aug 14, 2015 at 11:21 AM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/4146#discussion_r37086781>:
>
> > @@ -476,3 +479,12 @@ type BuildLogOptions struct {
> >  	// is not available yet. Otherwise the server will wait until the build has started.
> >  	NoWait bool
> >  }
> > +
> > +// SecretSpec specifies a secret to be included in a build pod and its corresponding mount point
> > +type SecretSpec struct {
>
> if we do decide this is limited to custom builds then we should move it
> into the custom build strategy as @liggitt <https://github.com/liggitt>
> originally suggested.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4146/files#r37086781>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift